### ISSUE TITLE
[Jepsen][YSQL] Enabling geo transactions lead to inconsistencies if tserver related nemeses used

### ISSUE THREAD

**Issue by qvad** [yugabyte/yugabyte-db#21116]
Jira Link: [DB-10148](https://yugabyte.atlassian.net/browse/DB-10148)
### Description

Running geo append tests with latest fixes on latest master/2.20 build leads to error

```
            :not #{:read-uncommitted},
            :also-not #{:consistent-view
                        :cursor-stability
                        :forward-consistent-view
                        :monotonic-atomic-view
                        :monotonic-snapshot-read
                        :monotonic-view
                        :read-committed
                        :repeatable-read
                        :serializable
                        :snapshot-isolation
                        :strong-serializable
                        :strong-session-serializable
                        :strong-session-snapshot-isolation
                        :strong-snapshot-isolation
                        :update-serializable}},
 :valid? false}


Analysis invalid! (ﾉಥ益ಥ）ﾉ ┻━┻
```

[20240220T080350.354Z.zip](https://github.com/yugabyte/yugabyte-db/files/14354263/20240220T080350.354Z.zip)

### Issue Type

kind/bug

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

[DB-10148]: https://yugabyte.atlassian.net/browse/DB-10148?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by qvad** [comment]
Slightly different error is in 2.20

```
                                             :type :G-single-realtime}]},
            :not #{:strong-snapshot-isolation},
            :also-not #{:strong-serializable}},
 :valid? false}


Analysis invalid! (ﾉಥ益ಥ）ﾉ ┻━┻
```

**Comment by rthallamko3** [comment]
Per @es1024 , Before backporting the fix for this, the fixes @20321 and #24317 should be backported otherwise the geo partitioning tests would fail lot more frequently.
